### PR TITLE
Restructure documentation to be built with documentation-builder

### PR DESCRIPTION
**Pull request by nottrobin**
This brings documentation format inline with our other documentation (ubuntu-core-docs, maas-docs etc.).

This is intended to be used with [this PR](https://github.com/ubuntudesign/docs.vanillaframework.io/pull/68) for now, and later to be pulled in by the docs.ubuntu.com application.

QA
---

``` bash
snap install documentation-builder
documentation-builder --base-directory docs --output-path docs/build
```

Open up the file generated at `docs/build/en/index.html` and browse around.

### CODE REVIEW

**Comment by anthonydillon** [comment]
@nottrobin we are targeting the `develop` branch now. Could you retarget please?

**Comment by anthonydillon** [comment]
Struggling to QA as:
```bash
$ documentation-builder --base-directory docs --output-path docs/build
zsh: command not found: documentation-builder
```
After installing documentation-builder.

File: docs/en/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    title: Typography
+  - location: base/code.md
+    title: Code
+  - location: base/forms.md
+    title: Forms
+  - location: base/reset.md
+    title: Reset
+  - location: base/lists.md
+    title: Lists
+  title: base
```
Comment (anthonydillon) [inline docs/en/metadata.yaml]: Can you capitalise please

File: docs/en/metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    title: Color
+  - location: settings/layout-settings.md
+    title: Layout
+  - location: settings/font-settings.md
+    title: Font
+  - location: settings/assets-settings.md
+    title: Assets
+  - location: settings/breakpoint-settings.md
+    title: Breakpoints
+  title: settings
```
Comment (anthonydillon) [inline docs/en/metadata.yaml]: Can you capitalise please

File: docs/en/metadata.yaml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    title: Code numbered
+  - location: patterns/buttons.md
+    title: Buttons
+  - location: patterns/lists.md
+    title: Lists
+  - location: patterns/links.md
+    title: Links
+  - location: patterns/pull-quote.md
+    title: Pull quote
+  title: patterns
```
Comment (anthonydillon) [inline docs/en/metadata.yaml]: Can you capitalise please

File: docs/en/metadata.yaml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    title: Clearfix
+  - location: utilities/off-screen.md
+    title: Off screen
+  - location: utilities/align.md
+    title: Align
+  - location: utilities/vertically-center.md
+    title: Vertically center
+  - location: utilities/equal-height.md
+    title: Equal height
+  title: utilities
```
Comment (anthonydillon) [inline docs/en/metadata.yaml]: Can you capitalise please

**Comment by anthonydillon** [comment]
This is good to land :+1:

**Comment by nottrobin** [comment]
[Change docs.vanillaframework.io to building with documentation-builder](https://trello.com/c/9hCRouMm/2533-change-docs-vanillaframework-io-to-building-with-documentation-builder)

**Pull request merged**